heksher/db_logic/util.py

```diff
@@ -0,0 +1,60 @@
+from typing import NamedTuple, Collection, Tuple, TypeVar, Generic, Sequence, Hashable, Dict, Union, Literal
+
+T = TypeVar('T', bound=Hashable)
+
+
+# pytype: disable=not-supported-yet
+
+class SupersequenceResults(NamedTuple, Generic[T]):
+    """
+    An object returned by is_supersequence to denote that the supersequence matches, along with the elements in the
+    supersequence that are missing from the subsequence
+    """
+    new_elements: Collection[Tuple[T, int]]
```

Comment:
Why did you make a new class for this?

Reply:
Clarity, I suppose, do you think it would be ok for `is_supersequence` to return `new_elements` directly?

Reply:
I don't think there's a difference between returning a complex object, and a complex object inside a class with a single variable. Makes it even less clear tbh.

You can create an alias for the long annotation and use it if you wish. Simple to both use and understand.